// Math/index.js
"use strict";
var Prelude = require("Prelude");
var abs = Math.abs;;
var acos = Math.acos;;
var asin = Math.asin;;
var atan = Math.atan;;
function atan2(y){  return function (x) {    return Math.atan2(y, x);  };};
var ceil = Math.ceil;;
var cos = Math.cos;;
var exp = Math.exp;;
var floor = Math.floor;;
var log = Math.log;;
function max(n1){  return function(n2) {    return Math.max(n1, n2);  }};
function min(n1){  return function(n2) {    return Math.min(n1, n2);  }};
function pow(n){  return function(p) {    return Math.pow(n, p);  }};
var round = Math.round;;
var sin = Math.sin;;
var sqrt = Math.sqrt;;
var tan = Math.tan;;
var e       = Math.E;;
var ln2     = Math.LN2;;
var ln10    = Math.LN10;;
var log2e   = Math.LOG2E;;
var log10e  = Math.LOG10E;;
var pi      = Math.PI;;
var sqrt1_2 = Math.SQRT1_2;;
var sqrt2   = Math.SQRT2;;
module.exports = {
    abs: abs, 
    acos: acos, 
    asin: asin, 
    atan: atan, 
    atan2: atan2, 
    ceil: ceil, 
    cos: cos, 
    e: e, 
    exp: exp, 
    floor: floor, 
    ln10: ln10, 
    ln2: ln2, 
    log: log, 
    log10e: log10e, 
    log2e: log2e, 
    max: max, 
    min: min, 
    pi: pi, 
    pow: pow, 
    round: round, 
    sin: sin, 
    sqrt: sqrt, 
    sqrt1_2: sqrt1_2, 
    sqrt2: sqrt2, 
    tan: tan
};
